apps/comps/hooks/useCompetitionBoosts.ts

```diff
@@ -0,0 +1,41 @@
+import { useInfiniteQuery } from "@tanstack/react-query";
+
+import { client } from "@/rpc/clients/client-side";
+import { getCompetitionPollingInterval } from "@/utils/competition-utils";
+
+/**
+ * Hook to fetch paginated boost allocations for a competition
+ * @param competitionId Competition ID
+ * @param limit Number of items per page
+ * @param enabled Whether the query is enabled
+ * @param competitionStatus Competition status for determining polling behavior
+ * @returns Infinite query result with boosts data and loadMore function
+ */
+export const useCompetitionBoosts = (
+  competitionId: string,
+  limit: number = 25,
+  enabled: boolean = true,
+  competitionStatus?: "active" | "pending" | "ending" | "ended",
+) => {
+  return useInfiniteQuery({
+    queryKey: ["competition", "boosts", competitionId, limit],
+    queryFn: async ({ pageParam = 0 }) => {
+      const result = await client.boost.competitionBoosts({
```

Comment:
Check out the other uses of our rpc via the tanstack client. It simplifies all this and returns the query options you pass to `useInfiniteQuery`.